addons/mail/static/src/components/activity/activity.xml

```diff
@@ -2,8 +2,8 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.Activity" owl="1">
-        <div class="o_Activity" t-attf-class="{{ className }}" t-on-click="activityView.onClickActivity" t-ref="root">
-            <t t-if="activityView">
+        <t t-if="activityView">
+            <div class="o_Activity" t-attf-class="{{ className }}" t-on-click="activityView.onClickActivity" t-ref="root">
```

Comment:
Note: this inversion should then be done on almost all our components, I'm not sure why this one in particular was done?
But the existing code should work too with Owl 2, I would propose to make these changes in a follow up PR instead.

Reply:
This one is definitely necessary, form views are broken without it

Reply:
Probably because `t-on-click="activityView.onClickActivity"` should have been `t-on-click="activityView and activityView.onClickActivity"` in the first place.

But from what I understand from other comments, if evaluation to `undefined` is no longer supported as handler (which I strongly disagree with), it's obviously not a solution to add the condition.

But then all of those occurrences that do have the condition should be changed to also move the `t-if` above them, regex such as `t-on.*?=".*?\band\b.*?"`

Reply:
I have restored the original order of the t-if and div and added the condition. I don't think the handler being undefined is a big problem since it will render as an empty div with no size and will not be clickable by a real user (the same is true for other components). Ideally, this component shouldn't be rendered at all when there is no activity view but this is a bigger refactoring and I'll leave that to you if that's something you would like to do later.